dandi/dandiapi.py

```diff
@@ -1341,19 +1341,22 @@ def get_content_url(
             raise NotFoundError(
                 "No matching URL found in asset's contentUrl metadata field"
             )
-        if follow_redirects is True:
-            url = self.client.request(
-                "HEAD", url, json_resp=False, allow_redirects=True
-            ).url
-        elif follow_redirects:
-            for _ in range(follow_redirects):
-                r = self.client.request(
-                    "HEAD", url, json_resp=False, allow_redirects=False
-                )
-                if "Location" in r.headers:
-                    url = r.headers["Location"]
-                else:
-                    break
+        try:
+            if follow_redirects is True:
+                url = self.client.request(
+                    "HEAD", url, json_resp=False, allow_redirects=True
+                ).url
+            elif follow_redirects:
+                for _ in range(follow_redirects):
+                    r = self.client.request(
+                        "HEAD", url, json_resp=False, allow_redirects=False
+                    )
+                    if "Location" in r.headers:
+                        url = r.headers["Location"]
+                    else:
+                        break
+        except requests.HTTPError as e:
+            url = e.request.url
```

Comment:
please add a unittest which would exercise this condition?  I guess it would be the one described in the original description of https://github.com/dandi/dandi-cli/issues/980 .

Reply:
Test added.